File: OwnHabits.Domain/Models/Skill.cs
using OwnHabits.Domain.Common;
using OwnHabits.Domain.Models.User;

namespace OwnHabits.Domain.Models;

public class Skill: BaseTextEntity
{
    public int Level { get; set; } = 1;
    public int CurrentExperience { get; set; } = 0;
    public int NextLevelExperience { get; set; } = 100;
    
    public virtual ICollection<SkillToCharacteristics> UpgradableCharacteristics { get; set; } = [];
    public virtual ICollection<GoalsToSkills> ExperiencedGoals { get; set; } = [];
    public virtual ICollection<UserToSkills> UserSkills { get; set; } = [];


    
}

File: Frontend/own-habits-frontend/src/types/Skill.ts
import type { GoalsToSkills } from './GoalsToSkills';
import type { SkillToCharacteristics } from './SkillToCharacteristics';
import type { UserToSkills } from './UserToSkills';

export interface Skill {
  id: string;
  userId: string;
  title: string;
  description?: string;
  level: number;
  currentExperience: number;
  nextLevelExperience: number;
  
  // Навигационные свойства
  upgradableCharacteristics: SkillToCharacteristics[];
  experiencedGoals: GoalsToSkills[];
  userSkills: UserToSkills[];
}